spec/virtual_includes_spec.rb

```diff
@@ -228,6 +240,64 @@
     end
   end
 
+  context ".merge_includes" do
+    it "merges when first is blank" do
+      first  = {}
+      second = {:key => {}}
+      result = {:key => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges when second is blank" do
+      first  = {:key => {}}
+      second = {}
+      result = {:key => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "keeps other keys" do
+      first  = {:other1 => {}}
+      second = {:other2 => {}}
+      result = {:other1 => {}, :other2 => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges blank, blank" do
+      first  = {:key => {}}
+      second = {:key => {}}
+      result = {:key => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges blank, hash" do
+      first  = {:key => {}}
+      second = {:key => {:more2 => {}}}
+      result = {:key => {:more2 => {}}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges hash, blank" do
+      first  = {:key => {:more1 => {}}}
+      second = {:key => {}}
+      result = {:key => {:more1 => {}}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges hash, hash - 1 level only" do
+      first  = {:key => {:more1 => {}}}
+      second = {:key => {:more2 => {}}}
+      result = {:key => {:more1 => {}, :more2 => {}}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges 2 overlapping hashes - 2 levels" do
+      first  = {:key => {:more => {:third => {:fourth1 => {}}}}}
+      second = {:key => {:more => {:third => {:fourth2 => true}}}}
+      result = {:key => {:more => {:third => {:fourth1 => {}, :fourth2 => true}}}}
```

Comment:
Can you add a test like this but where the deeply nested hash keys overlap, to demonstrate who wins?

```
      first  = {:key => {:more => {:third => {:fourth => {}}}}}
      second = {:key => {:more => {:third => {:fourth => true}}}}
      result = {:key => {:more => {:third => {:fourth => ????}}}
```

Reply:
2nd commit added this